src/xds/client.rs

```diff
@@ -447,6 +479,19 @@ impl AdsClient {
             .fields
             .append(&mut Self::build_struct(self.config.proxy_metadata.clone()).fields);
 
+        // Lookup ISTIO_METAJSON_* environment variables and add them to the node metadata
+        for (key, val) in std::env::vars().filter(|(key, _)| key.starts_with(ISTIO_METAJSON_PREFIX))
```

Comment:
nit: the other `env` reading is done in `config.rs`. Should this go there for consistency and be passed in on another field `metadata_json`?

Reply:
Couldn't add it to ProxyConfig because of prost types not implementing some traits. That, and it made sense to put it in `node()` because that's where we are reading node metadata other than proxy_metadata, such as name, namespace, instance IPs, etc.